specs/online_order_spec.rb

```diff
@@ -3,69 +3,137 @@
 require 'minitest/skip_dsl'
 
 # TODO: uncomment the next line once you start wave 3
-# require_relative '../lib/online_order'
+require_relative '../lib/onlineorder'
 # You may also need to require other classes here
 
 # Because an OnlineOrder is a kind of Order, and we've
 # already tested a bunch of functionality on Order,
 # we effectively get all that testing for free! Here we'll
 # only test things that are different.
 
-xdescribe "OnlineOrder" do
+describe "OnlineOrder" do
   describe "#initialize" do
     it "Is a kind of Order" do
-      # Check that an OnlineOrder is in fact a kind of Order
-
+      id = 1
+	  products = {'socks' => 2.34}
+	  customer = 5
+	  status = 'paid'
       # Instatiate your OnlineOrder here
-      # online_order =
-      # online_order.must_be_kind_of Grocery::Order
+      online_order = Grocery::OnlineOrder.new(id, products, customer, status)
+      online_order.must_be_kind_of Grocery::OnlineOrder
     end
 
-    it "Can access Customer object" do
-      # TODO: Your test code here!
+	it "Can access Customer object" do
+		id = 1
+		products = {'socks' => 2.34}
+		customer = 5
+		status = 'paid'
+		online_order = Grocery::OnlineOrder.new(id, products, customer, status)
+
+		online_order.must_respond_to :customer
+		online_order.customer.must_equal customer
+		online_order.customer.must_be_kind_of Integer
     end
 
     it "Can access the online order status" do
-      # TODO: Your test code here!
+		id = 1
+		products = {'socks' => 2.34}
+		customer = 5
+		status = 'paid'
+		online_order = Grocery::OnlineOrder.new(id, products, customer, status)
+		
+		online_order.must_respond_to :status
+		online_order.status.must_equal status
+		online_order.status.must_be_kind_of String
     end
   end
 
   describe "#total" do
     it "Adds a shipping fee" do
-      # TODO: Your test code here!
+		id = 1
+		products = {'socks' => 2.34}
+		customer = 5
+		status = 'paid'
+		online_order = Grocery::OnlineOrder.new(id, products, customer, status)
+		
+		calculated_total = online_order.total
+		raw_total = (2.34*1.075).round(2) + 10
+		
+		calculated_total.must_equal raw_total
     end
 
     it "Doesn't add a shipping fee if there are no products" do
-      # TODO: Your test code here!
+		id = 1
+		products = {}
+		customer = 5
+		status = 'pending'
+		online_order = Grocery::OnlineOrder.new(id, products, customer, status)
+		
+		total = online_order.total
+		
+		total.must_equal 0
     end
   end
 
   describe "#add_product" do
     it "Does not permit action for processing, shipped or completed statuses" do
-      # TODO: Your test code here!
+		id = 1
```

Comment:
This test doesn't actually run the `add_product` method! You should call that method and check that it either raises an error or returns a value to indicate it didn't work.